### ISSUE TITLE
MVVM: Remove reference to MainPageViewModel from HorizontalMultitaskingControl

### ISSUE THREAD

**Issue by yaichenbaum** [files-community/Files#4221]
We can use dependency properties instead of accessing the view model directly from the control.

**Comment by 0x5bfa** [comment]
Do you remember why did you come up with this idea?

`HorizontalMultitaskingControl : BaseMultitaskingControl` certainly does have a reference to the MainPageViewModel. But I think it is not a problem since it is injected by IOC.

```cs
protected readonly MainPageViewModel mainPageViewModel = Ioc.Default.GetRequiredService<MainPageViewModel>();
```

**Comment by yaira2** [comment]
The control is supposed to be modular without any knowledge of the apps logic.

**Comment by 0x5bfa** [comment]
This reference is used for reopening recently closed tabs.

To fix this issue, I seem to have to move adding tabs methods from MainPageViewModel to a statis helper.

**Comment by yaira2** [comment]
Where exactly is it used, in the right click menu item for "reopen closed tab"?

**Comment by 0x5bfa** [comment]
Yes. There is a method called ReopenRecentlyClosedTabs() in BaseTabBar class.

**Comment by yaira2** [comment]
Perhaps a solution in the meantime is to reference the action.
